cl/users/models.py

```diff
@@ -174,6 +205,86 @@ class Meta:
         verbose_name_plural = "user profiles"
 
 
+class EmailFlag(AbstractDateTimeModel):
+    """Stores flags for email addresses
+
+    Two types of :object_type:
+    BAN: ban an email address and avoid sending any email
+    FLAG: flag an email address for a special treatment e.g small_email_only.
+    :flag: the actual flag assigned, like: small_email_only
+    :event_sub_type: the SNS bounce subtype that triggered the object
+    :email_address: an EmailFlag object belongs to an email address instead
+    of a user, in this way, if users change their email address this won't
+    affect new user email address.
```

Comment:
For the field level details, any reason to put it here instead of in the help_text or in a comment near the field? I think it moves the comment further from where you need it. It's also not how we usually have done it, but if this is a pattern elsewhere, I'd be happy to consider a different approach.